Userland/Libraries/LibGfx/ImageFormats/GIFWriter.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/BitStream.h>
+#include <LibCompress/Lzw.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/ImageFormats/GIFWriter.h>
+
+namespace Gfx {
+
+namespace {
+
+ErrorOr<void> write_header(Stream& stream)
+{
+    // 17. Header
+    TRY(stream.write_until_depleted("GIF87a"sv));
+    return {};
+}
+
+ErrorOr<void> write_logical_descriptor(BigEndianOutputBitStream& stream, Bitmap const& bitmap)
+{
+    // 18. Logical Screen Descriptor
+
+    if (bitmap.width() > NumericLimits<u16>::max() || bitmap.height() > NumericLimits<u16>::max())
+        return Error::from_string_literal("Bitmap size is too big for a GIF");
+
+    TRY(stream.write_value<u16>(bitmap.width()));
+    TRY(stream.write_value<u16>(bitmap.height()));
+
+    // Global Color Table Flag
+    TRY(stream.write_bits(true, 1));
+    // Color Resolution
+    TRY(stream.write_bits(6u, 3));
+    // Sort Flag
+    TRY(stream.write_bits(false, 1));
+    // Size of Global Color Table
+    TRY(stream.write_bits(7u, 3));
+
+    // Background Color Index
+    TRY(stream.write_value<u8>(0));
+
+    // Pixel Aspect Ratio
+    // NOTE: This is not mandatory.
+    TRY(stream.write_value<u8>(0));
```

Comment:
Question for dummies: How does it know whether the pixel aspect ratio is there or not?

Reply:
The byte is always here in the file, if that's your question (I will clarify the comment as well). Most decoders simply discard the value.